template.tpl

```diff
@@ -2139,7 +2139,8 @@ switch (data.eventType) {
     const commands =
       data.customCommandTable && data.customCommandTable.length ? data.customCommandTable : [];
     commands.forEach((c) => {
-      tracker(mkCommand(c.name), c.args);
+      const commandArgs = c.args === '' ? undefined : c.args;
```

Comment:
This change makes it impossible to use empty string as an actual value... but I think this is an OK default for the better UX in terms of POLA for the majority of users.

Of the commands that can take a string as a parameter, I don't really see any place where an empty string is actually desirable/useful (it will usually just fallback to a default alternative value, which `undefined` should do as well). _Maybe_ some people who _really_ want empty string for user ID or something, but that's a weird requirement.

Technically there's a dedicated `undefined` variable in GTM that's probably the "right" way for the user to specify this, but it's pretty niche so we shouldn't expect people to understand why that's needed and to implement it. Especially since without this change the empty string can cause 'do nothing silently' behaviour (e.g. for `updatePageActivity`), so people might never know this is a problem at the moment. 

I think there's a tiny API break here for tracker callbacks, but due to how `mkCommand` works I don't think they work via the template anyway, so that's fine. (Additional arguments after the function are passed to the callback as arguments, so if you were expecting empty string you wouldn't get it anymore)

Reply:
Thank you for reviewing @jethron !
You are right, this is a great point. I've been trying to think of alternatives.

What if, instead of disallowing empty strings as arguments (as this PR does), we just enforce a validation rule on the `Command Argument` column for the value to be non-empty. That way, on one hand, we do enforce people to specify an argument (e.g. for advanced or edge-case requirements, a variable that evaluates to empty string, if desired, or to undefined, if needed), but on the other hand, this will not be a silent behaviour.

Would that make sense?

Reply:
Oh yeah, that sounds like a great alternative! At least then we can spell out the ambiguity in the error and explain that an `undefined` variable is required if you explicitly don't want a value.